tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -98,34 +100,285 @@ def sections(
+                Section.Code(code=Op.RETF + Op.STOP, code_outputs=0),
+            ]
+        case Op.JUMPF:
+            sections = [
+                Section.Code(code=Op.JUMPF[1] + Op.STOP),
+                Section.Code(code=Op.STOP),
+            ]
+        case Op.RETURNCONTRACT:
+            sections = [
+                Section.Code(code=Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+                Section.Container(
+                    container=Container(
+                        sections=[
+                            Section.Code(code=Op.RETURNCONTRACT[0](0, 0) + Op.STOP),
+                            Section.Container(Container.Code(code=Op.STOP)),
+                        ]
+                    )
+                ),
+            ]
+        case Op.RETURN | Op.REVERT | Op.INVALID:
+            sections = [
+                Section.Code(code=Op.PUSH1(0) + Op.PUSH1(0) + opcode + Op.STOP),
+            ]
+        case _:
+            raise NotImplementedError(f"Opcode {opcode} is not implemented")
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.UNREACHABLE_INSTRUCTIONS
+        if opcode != Op.RETURNCONTRACT
+        else EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    sorted(op for op in valid_eof_opcodes if op.min_stack_height > 0)
+    + [
+        # Opcodes that have variable min_stack_height
+        Op.SWAPN[0x00],
+        Op.SWAPN[0xFF],
+        Op.DUPN[0x00],
+        Op.DUPN[0xFF],
+        Op.EXCHANGE[0x00],
+        Op.EXCHANGE[0xFF],
+    ],
+)
+def test_all_opcodes_stack_underflow(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test stack underflow on all opcodes that require at least one item on the stack
+    """
+    sections: List[Section]
+    if opcode == Op.EOFCREATE:
+        sections = [
+            Section.Code(code=Op.PUSH0 * (opcode.min_stack_height - 1) + opcode[0] + Op.STOP),
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            ),
+        ]
+    elif opcode == Op.RETURNCONTRACT:
+        sections = [
+            Section.Code(code=Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.PUSH0 * (opcode.min_stack_height - 1) + opcode[0]),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            ),
+        ]
     else:
-        eof_code = Container(sections=sections)
+        bytecode = Op.PUSH0 * (opcode.min_stack_height - 1)
+        if opcode.has_data_portion():
```

Comment:
Oh, this is even better. Instead of modifying opcode it makes bytecode which construction is much cleaner